Dockerfile

```diff
@@ -0,0 +1,49 @@
+#########################################################
+# Build the sources and provide the result in a multi stage
+# docker container. The alpine build image has to match
+# the alpine image in the referencing runtime container.
+#########################################################
+FROM golang:1.10.1-alpine3.7 AS build-env
+
+# Set all env variables needed for go
+ENV GOBIN /go/bin
+ENV GOPATH /go
+
+# We need so that dep can fetch it's dependencies
+RUN apk --no-cache add git
+
+# Directory in workspace
+RUN mkdir -p "/go/src/github.com/Peripli/service-manager"
+COPY . "/go/src/github.com/Peripli/service-manager"
+WORKDIR "/go/src/github.com/Peripli/service-manager"
+
+# Install dep, dependencies and build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go get github.com/golang/dep/cmd/dep && \
+    dep ensure -v && \
+    go build -o /main cmd/main.go
+
+# RUN go build -o /main cmd/main.go
+
+########################################################
+# Build the runtime container
+########################################################
+FROM alpine:3.7
+WORKDIR /app
+
+# Copy the executable file
+COPY --from=build-env /main /app/
+
+# Copy the application.yml file to the executable container
+COPY --from=build-env /go/src/github.com/Peripli/service-manager/application.yml /app/
+
+# Create folder for migration scripts
+RUN mkdir -p "/app/storage/postgres/migrations"
+
+# Copy migration scripts
+COPY --from=build-env /go/src/github.com/Peripli/service-manager/storage/postgres/migrations/ /app/storage/postgres/migrations/
+
+ARG SM_RUN_ENV=local
```

Comment:
If we remove it then, for CF we would have to set the environment in the manifest when the container is being pushed. Other way would be to set the default to be "cf". Suggestions are welcome. Generally your idea is better than it is now.

Reply:
Default local seems best in my opinion.

I have one more idea -  have the run_mode be a cmd flag and be passed as last thing in docker run command (cmd arg) - as we know when we have an ENTRYPOINT in the Dockefile, everything passed to the docker run command after the image name is considered CMD args appended to the entrypoint.  This will make the dockerfile cleaner (we can remove the env var)
However, I am not sure how to pass this cmd args in the manifest.